### ISSUE TITLE
datadog is not one of the available choices

### ISSUE THREAD

**Issue by mathieux51** [sleuth-io/terraform-provider-sleuth#163]
### Terraform Version

```bash
t -version
OpenTofu v1.6.2
on darwin_arm64
+ provider registry.opentofu.org/hashicorp/aws v5.46.0
+ provider registry.opentofu.org/sleuth-io/sleuth v0.4.7
```

### Affected Resource(s)

- `sleuth_metric_impact_source`
### Terraform Configuration Files
```hcl
resource "sleuth_metric_impact_source" "ingress" {
  project_slug     = "my-project"
  environment_slug = "production"
  name             = "my-service"
  provider_type    = "datadog"
  query            = "sum:nginx_ingress.controller.requests{service:my-service}"
  less_is_better   = false
}
```

### Error Output
```
Error: Errors creating impact source:  [{Field:auth Messages:[Select a valid choice. datadog is not one of the available choices.]}]
```

### Expected Behavior
[According to the documentation](https://registry.terraform.io/providers/sleuth-io/sleuth/latest/docs/resources/metric_impact_source), resource should be created

### Actual Behavior
Fails

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply -auto-approve`

**Comment by mathieux51** [comment]
Not sure who to ping, seems like @JureMalovrh and @zigomir were the last contributors

**Comment by recodedecode** [comment]
I am having the same issue with `newrelic` as the `provider_type` using `sleuth-io/sleuth` version `0.3.0`. I believe it might be an error with the API response and not this package as looking at the [Go code](https://github.com/sleuth-io/terraform-provider-sleuth/blob/main/internal/provider/resource_metric_impact_source.go#L43) it doesn't appear there is any validation against an actual enum. The [GraphQL ImpactProvider (see query here)](https://app.sleuth.io/graphql#query=query%20%7B%0A%20%20__type(name%3A%22ImpactProvider%22)%20%7B%0A%09%09name%0A%20%20%20%20enumValues%7B%0A%20%20%20%20%20%20name%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) does list both `newrelic` and `datadog` as valid options.

**Comment by mathieux51** [comment]
@recodedecode The README clearly states there is no support for `newrelic`: 
> No support for New Relic as impact source provider
whereas `datadog` [is in the Terraform provider documentation](https://registry.terraform.io/providers/sleuth-io/sleuth/latest/docs/resources/metric_impact_source) as an example for `sleuth_metric_impact_source`.

**Comment by recodedecode** [comment]
Ok, so I missed that fair call out. However, it's in the [docs](https://help.sleuth.io/integrations-1/impact-sources/metrics/newrelic) it's in the [GraphQL API](https://app.sleuth.io/graphql#query=query%20%7B%0A%20%20__type(name%3A%22ImpactProvider%22)%20%7B%0A%09%09name%0A%20%20%20%20enumValues%7B%0A%20%20%20%20%20%20name%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D) (hit the play button) as an apparently valid impact provider and having used this provider for other sources I just tried to tie this one in too.

**Comment by JureMalovrh** [comment]
@mathieux51 were you able to solve the issue?

**Comment by recodedecode** [comment]
I was able to get this working with your direction @JureMalovrh but it wasn't obvious as I could not initially find where the appropriate `integration_slug` value was. I did eventually find it by looking at data returned from the GraphQL API over the Sleuth website network requests for our organisation.

With this I was actually able to create a New Relic impact source. So while it says in the docs on this repo that you can't use New Relic with this provider, it was in-fact possible. I believe this is the case because the provider is simply calling the Sleuth API and performing no checks on the enum internally within the provider.

Hope this helps the next person.

**Comment by JureMalovrh** [comment]
Yeah, we have a ticket open for exposing integration slugs via UI to enable easier copying and usage. I will close the ticket now and bump the priority of the internal ticket.

**Comment by zigomir** [comment]
Closing since Datadog has landed with https://github.com/sleuth-io/terraform-provider-sleuth/pull/94
